utils/shared/src/main/scala/fastparse/UberBuffer.scala

```diff
@@ -0,0 +1,103 @@
+package fastparse
+
+import scala.reflect.ClassTag
+
+/**
+  * A very fast circular, growable read-write byte buffer.
+  */
+class UberBuffer[T: ClassTag](initSize: Int = 32){ self =>
```

Comment:
`ClassTag` is necessary to instantiate a generic `new Array[T]` http://stackoverflow.com/a/16921379/871202